src/mo_frontend/typing.ml

```diff
@@ -2970,7 +2970,8 @@ and check_parenthetical env typ_opt = function
        let s, _, _, _, ts2 = T.as_func fun_ty in
```

Comment:
check_parenthetical seems to only reject application that return an async*. Does it complain when the parenthetical is attached to an async* expression ?

e.g.
```
(with cycles = 100) async* { };
```

Reply:
Odd, because I'm sure I commented on this earlier but perhaps I didn't submit the comment properly. Or GH lost it.